weighted_levenshtein/clev.pyx

```diff
@@ -132,11 +132,20 @@ cdef inline DTYPE_t row_insert_range_cost(
 
 # End Array2D
 
+cdef unsigned int* convert_string_to_int_array(unsigned char* str, Py_ssize_t size):
```

Comment:
I believe this function can be `nogil` if we get rid of `enumerate` and `ord`:

```
    cdef unsigned int* intarr = <unsigned int*> malloc(size * sizeof(unsigned int))
    for i in range(size):
        intarr[i] = str[i]
    return intarr
```

I don't think `ord` is necessary since in C `unsigned char` can be widened to `unsigned int` implicitly without loss of precision.